main/main.c

```diff
@@ -45,6 +46,59 @@ uint64_t cycles_per_sec;
 uint64_t cycles_per_ms;
 uint64_t picosec_per_cycle;
 
+char *lua_base_dir = NULL;
+char *gatekeeper_config_file = NULL;
+
+/* Argp's global variables. */
+const char *argp_program_version = "gatekeeper 1.0";
+
+/* Arguments. */
+static char adoc[] = "<LUA_BASE_DIR> <GATEKEEPER_CONFIG_FILE>";
+
+static char doc[] = "gatekeeper -- the first open source "
+	"DDoS protection system";
+
+static struct argp_option options[] = {
+	{"lua-base-dir", 'd', "DIR", 0,
+		"Base directory DIR for Gatekeeper Lua files", 0},
+	{"gatekeeper-config-file", 'f', "FILE", 0,
+		"Lua configuration FILE to initialize Gatekeeper", 0},
+	{ 0 }
+};
+
+struct args {
+	char *lua_base_dir;
+	char *gatekeeper_config_file;
+};
+
+static error_t
+parse_opt(int key, char *arg, struct argp_state *state)
+{
+	struct args *args = state->input;
+
+	switch (key) {
+	case 'd':
+		args->lua_base_dir = arg;
+		break;
+
+	case 'f':
+		args->gatekeeper_config_file = arg;
+		break;
+
+	case ARGP_KEY_INIT:
+		args->lua_base_dir = NULL;
+		args->gatekeeper_config_file = NULL;
```

Comment:
Once `args->lua_base_dir` and `args->gatekeeper_config_file` have default values, we can drop the two assignments above.